src/ROOTNTupleWriter.cc

```diff
@@ -0,0 +1,209 @@
+#include "podio/ROOTNTupleWriter.h"
+#include "podio/GenericParameters.h"
+#include "rootUtils.h"
+#include "podio/CollectionBase.h"
+#include "podio/podioVersion.h"
+
+#include "TFile.h"
+#include <ROOT/RNTuple.hxx>
+#include <ROOT/RNTupleModel.hxx>
+
+#include <algorithm>
+
+namespace podio {
+
+ROOTNTupleWriter::ROOTNTupleWriter(const std::string& filename) :
+  m_metadata(ROOT::Experimental::RNTupleModel::Create()),
+  m_file(new TFile(filename.c_str(), "RECREATE", "data file"))
+  {
+  }
+
+ROOTNTupleWriter::~ROOTNTupleWriter() {
+  if (!m_finished) {
+    finish();
+  }
+}
+
+template<typename T>
+void ROOTNTupleWriter::fillParams(const std::string& category, GenericParameters& params) {
+  auto gpKeys = m_writers[category]->GetModel()->Get<std::vector<std::string>>(root_utils::getGPKey<T>());
+  auto gpValues = m_writers[category]->GetModel()->Get<std::vector<std::vector<T>>>(root_utils::getGPValue<T>());
+  gpKeys->clear();
+  gpValues->clear();
```

Comment:
Maybe also add a `reserve(params.getMap<T>().size()` here? (Although it probably doesn't matter too much apart from the first event).

Reply:
Done, yeah I think it would not be measurable unless you have lots and lots (4 or 5 digits?) of collections